specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+                        CHECK_FAILURE(certificateList->get_Count(&certificateListCount));
+
+                        ICoreWebView2ClientCertificate* certificate = nullptr;
+
+                        ClientCertificate client_certificate;
+                        for (UINT i = 0; i < certificateListCount; i++)
+                        {
+                            CHECK_FAILURE(certificateList->GetValueAtIndex(i, &certificate));
+
+                            wil::unique_cotaskmem_string subject;
+                            CHECK_FAILURE(certificate->get_Subject(&subject));
+                            client_certificate.Subject = subject.get();
+
+                            wil::unique_cotaskmem_string displayName;
+                            CHECK_FAILURE(certificate->get_DisplayName(&displayName));
+                            client_certificate.DisplayName = displayName.get();
+
+                            wil::unique_cotaskmem_string issuer;
+                            CHECK_FAILURE(certificate->get_Issuer(&issuer));
+                            client_certificate.Issuer = issuer.get();
+
+                            wil::unique_cotaskmem_string serialNumber;
+                            CHECK_FAILURE(certificate->get_SerialNumber(&serialNumber));
+
+                            COREWEBVIEW2_CLIENT_CERTIFICATE_KIND certificateKind;
+                            CHECK_FAILURE(certificate->get_CertificateKind(&certificateKind));
+                            client_certificate.CertificateKind = NameOfCertificateKind(certificateKind);
+
+                            double startDate;
+                            CHECK_FAILURE(certificate->get_ValidStartDate(&startDate));
+                            client_certificate.StartDate = startDate;
+
+                            double expiryDate;
+                            CHECK_FAILURE(certificate->get_ValidExpiryDate(&expiryDate));
+                            client_certificate.ExpiryDate = expiryDate;
+
+                            client_certificates_.push_back(client_certificate);
+                        }
+
+                        // Display custom dialog box for the client certificate selection.
+                        ClientCertificateSelectionDialog dialog(
+                            m_appWindow->GetMainWindow(), L"Select a Certificate for authentication", host.get(), port, client_certificates_);
+
+                        if (dialog.confirmed)
+                        {
+                            int selectedCertificate = dialog.selectedItem;
+                            if (selectedCertificate >= 0)
+                            {
+                                CHECK_FAILURE(
+                                    certificateList->GetValueAtIndex(selectedCertificate, &certificate));
+
+                                CHECK_FAILURE(args->put_SelectedCertificate(certificate));
+                                CHECK_FAILURE(args->put_ClientCertificateRequestResponseState(
```

Comment:
I don't think the `State` suffix is adding any value. Actually, I don't think `ClientCertificateRequest` is either. So just `Response` is good.